packages/dds/tree/src/test/simple-tree/arrayNode.spec.ts

```diff
@@ -53,6 +53,76 @@ describe("ArrayNode", () => {
 			const array = hydrate(schemaType, jsArray);
 			assert.equal(JSON.stringify(array), JSON.stringify(jsArray));
 		});
+
+		it("removeAt()", () => {
+			const array = hydrate(schemaType, [0, 1, 2]);
+			array.removeAt(1);
+			assert.deepEqual([...array], [0, 2]);
+		});
+
+		describe("removeRange", () => {
+			it("no arguments", () => {
+				const jsArray = [0, 1, 2];
+				const array = hydrate(schemaType, jsArray);
+				assert.equal(array.length, 3);
+				array.removeRange();
+				assert.equal(array.length, 0);
+				assert.deepEqual([...array], []);
+			});
+
+			it("empty array no arguments", () => {
+				const array = hydrate(schemaType, []);
+				array.removeRange();
+			});
+
+			it("middle", () => {
+				const list = hydrate(schemaType, [0, 1, 2, 3]);
+				list.removeRange(/* start: */ 1, /* end: */ 3);
+				assert.deepEqual([...list], [0, 3]);
+			});
+
+			it("all", () => {
+				const list = hydrate(schemaType, [0, 1, 2, 3]);
+				list.removeRange(0, 4);
+				assert.deepEqual([...list], []);
+			});
+
+			it("past end", () => {
+				const list = hydrate(schemaType, [0, 1, 2, 3]);
+				list.removeRange(1, Infinity);
+				assert.deepEqual([...list], [0]);
+			});
+
+			it("empty range", () => {
+				const list = hydrate(schemaType, [0, 1, 2, 3]);
+				list.removeRange(2, 2);
+				assert.deepEqual([...list], [0, 1, 2, 3]);
+			});
+
+			it("empty range - at start", () => {
+				const list = hydrate(schemaType, [0, 1, 2, 3]);
+				list.removeRange(0, 0);
+				assert.deepEqual([...list], [0, 1, 2, 3]);
+			});
+
+			it("empty range - at end", () => {
+				const list = hydrate(schemaType, [0, 1, 2, 3]);
+				list.removeRange(4, 4);
+				assert.deepEqual([...list], [0, 1, 2, 3]);
+			});
+
```

Comment:
I think it'd be worth testing and locking in the behavior of three more cases:

* "empty range before start" (e.g. `removeRange(-1, -1)`)
* "empty range after end" (e.g. `removeRange(5, 5)`)
* "'empty' infinity range" (e.g. `removeRange(Infinity, Infinity)`)

Reply:
those last two are just more versions of start past end of array ( the first case ) and the first one will just hit the same validation as the existing negative index test.

Since we validate the inputs before we do anything depending on the length of the range, that doesn't actually add any coverage, as they all just fail validation of the start in ways we already have test for before even looking at the end value.

Reply:
I agree that they don't add any additional coverage given the current implementation. But they might be more resilient against other, future implementations if we ever change things. For example, you could imagine somebody adding an optimization at the top of the method like "if (start === end) { return }", but we'd want them to know that they changed the behavior of the "before start" and "after end" cases. Handling all the combinations of indexes can be fiddly (our team has gotten it wrong a couple times now), so I'm just being a little conservative.